vowpalwabbit/v_array.h

```diff
@@ -54,13 +57,19 @@ template<class T> struct v_array
     _end++;
   }
   T& operator[](size_t i) const { return _begin[i]; }
+  T& get(size_t i) const { return _begin[i]; }
   inline size_t size() const {return _end-_begin;}
   void resize(size_t length)
   { if ((size_t)(end_array-_begin) != length)
     { size_t old_len = _end-_begin;
       T* temp = (T *)realloc(_begin, sizeof(T) * length);
       if ((temp == nullptr) && ((sizeof(T)*length) > 0))
-      { THROW("realloc of " << length << " failed in resize().  out of memory?");
+      { 
+#ifdef VW_NOEXCEPT
+	return;
```

Comment:
Won't this be a silent fail?

Reply:
yes it will, but this would have required a huge amount of change in the codebase

---
In reply to: [188081473](https://github.com/JohnLangford/vowpal_wabbit/pull/1481#discussion_r188081473) [](ancestors = 188081473)

Reply:
Yes.  True.  Given time constraints, one way to move forward might be to convince ourselves that in the limited ways that we expect this code to get exercised in the near future, we are safe.  Add a work item to fix this and schedule it properly.  Probably warrants a face to face discussion with @jl@hunch.net .

---
In reply to: [188103898](https://github.com/JohnLangford/vowpal_wabbit/pull/1481#discussion_r188103898) [](ancestors = 188103898,188081473)